src/bin/inx-chronicle/stardust_inx/mod.rs

```diff
@@ -252,19 +273,43 @@ impl HandleEvent<Result<LedgerUpdateRecord, InxError>> for InxWorker {
 
 #[instrument(skip_all, fields(num = outputs.len()), level = "trace")]
 async fn insert_unspent_outputs(db: &MongoDb, outputs: Vec<LedgerOutput>) -> Result<(), InxError> {
-    try_join!(
-        db.insert_unspent_outputs(outputs.iter()),
-        db.insert_unspent_ledger_updates(outputs.iter())
-    )?;
+    let output_collection = db.collection::<OutputCollection>();
+    let ledger_collection = db.collection::<LedgerUpdateCollection>();
+    try_join! {
+        async {
+            for batch in &outputs.iter().chunks(INSERT_BATCH_SIZE) {
+                output_collection.insert_unspent_outputs(batch).await?;
+            }
+            Result::<_, InxError>::Ok(())
```

Comment:
The compiler can't resolve the exact Result, but for the other async block it can?

Reply:
Yes, because of this one.